lib/rbeapi/api/ipinterfaces.rb

```diff
@@ -352,6 +379,58 @@ def set_helper_addresses(name, opts = {})
       end
 
       ##
+      # set_secondary_addresses configures the list of secondary addresses on the ip
+      # interface. An IP interface can have one or more secondary addresses
+      # configured. If no value is provided, the secondary address configuration
+      # is set using the no keyword. If the default option is specified and
+      # set to true, then the secondary address values are defaulted using the
+      # default keyword.
+      #
+      # @since eos_version ?.??.?
+      #
+      # ===Commands
+      #   interface <name>
+      #     ip address <value> secondary
+      #     no ip address <value> secondary
+      #
+      # @param name [String] The name of the interface to configure the
+      #   address in the node. The name must be the full interface name.
+      #
+      # @param opts [Hash] Optional keyword arguments.
+      #
+      # @option opts value [Array] The list of IP addresses to configure as
+      #   secondary address on the interface. The secondary addresses must be valid
+      #   addresses in the main interface's subnet.
+      #
+      # @option opts default [Boolean] Configure the ip secondary address values
+      #    using the default keyword.
+      #
+      def set_secondary_addresses(name, opts = {})
+        value = opts[:value]
+        enable = opts.fetch(:enable, true)
+        default = opts[:default] || false
+
+        if value
+          fail ArgumentError, 'value must be an Array' unless value.is_a?(Array)
+        end
+
+        case default
+        when true
+          cmds = [ ]
+        when false
+          cmds = [ ]
+          matches = config.scan(/(?<=^\s{3}ip\saddress\s)(.+)(\ssecondary)$/)
+          response = []
+          matches.each do |ip|
+            response << ip[0]
+          end
+          response.each { |addr| cmds << "no ip address #{addr} secondary" } if enable
+          value.each { |addr| cmds << "ip address #{addr} secondary" } if enable
```

Comment:
How can lines 427 and 428 both have if enable? Should't one be for the case of 'not enable'?